src/main/java/org/myrobotlab/kinematics/IMEngine.java

```diff
@@ -583,8 +583,8 @@ public void updateLinksPosition(ServoData data) {
     for (DHLink l : arm.getLinks()) {
       if (l.getName().equals(data.name)) {
         l.addPositionValue(data.pos);
-        l.setState(data.state);
-        l.setVelocity(data.velocity);
+        // l.setState(data.state);
+        l.setVelocity(data.speed);

```

Comment:
internal I'm still of the belief setSpeed is correct - previously setSpeed was the wrong implementation (mine) but has since been corrected.  velocity is the incorrect name.